### DISCUSSION TITLE
关于搭建Web开放平台|Android app|C#重写的需求讨论

### DISCUSSION THREAD

**Discussion by Waldenth** [Waldenth/FuckHamSports#5] [Ideas]
陆续有人反应环境配置容易和原有的配置的冲突、只能PC端操作等有很多不便的地方。现在手里有一台空闲流量很大的轻量应用服务器，搭建一个开放性Web平台，将服务转移到服务器上运行应该是可行的。Android app目前同样也支持Java开发(虽然有点过时了但也能用)，C#重写的话能彻底摆脱JDK的依赖，但是感觉没什么人有这个需求。

大家对重构有需求么？

**Comment by 18209618470** [comment]
大佬，能不能把那个跑步的软件里面的经纬度设一下就不用在改的那种，要不然每次一重新打开经纬度又回到默认的那个了，，谢谢大佬！

**Reply by 18209618470** [reply]
忘大佬能回复我，谢谢！

**Reply by Waldenth** [reply]
我已经在考虑这个问题，目前的想法是设计一个保存设置的配置文件，启动时尝试读取。我尽量早点实现你的这个功能需求.

**Reply by 18209618470** [reply]
大佬，加油！争取弄出好的软件出来。

**Reply by Waldenth** [reply]
只是个娱乐小项目而已，目前已经更新0.1.5版本，应该基本实现了你的需求，你可以测试一下。

**Reply by 18209618470** [reply]
大哥，为啥保存配置失败，是怎么回事啊？
![Uploading IMG_20210522_150456.jpg…]()

**Reply by Waldenth** [reply]
有问题请提交issue或者说明复现环境以及操作步骤，给出结果图和二进制文件版本，过于简单的描述无法复现你的问题

**Reply by 18209618470** [reply]
大佬，我冒昧的问一句，阳光体育的后台能不能检测出来啊，希望不要介意！

**Reply by Waldenth** [reply]
> 大佬，我冒昧的问一句，阳光体育的后台能不能检测出来啊，希望不要介意！

如果不使用时间休眠，由于开始和结束的请求时间相差太短，后台很容易发现数据异常。即使休眠也不能保证完全避开检测，我收到过有人因为一天内多次频繁请求而遭到遭到封禁的报告。

**Reply by 18209618470** [reply]
大佬，那像那些网上的代跑是怎么避免被检测到的呢？还有就是如果使用休眠的话会很慢很慢。

**Reply by Waldenth** [reply]
> 大佬，那像那些网上的代跑是怎么避免被检测到的呢？还有就是如果使用休眠的话会很慢很慢。

网上绝大多数代跑原理与本项目完全相同，甚至更容易被检测，因为服务器IP固定且进行了大量请求且没有进行模拟休眠，没有被检测只是因为阳光体育没有进行过日志检查。唯一可能完全避免检测的方式是使用root权限进行虚拟定位，而网上的代跑显然不可能帮你做这些。

**Reply by 18209618470** [reply]
大佬能在这个项目中实现虚拟定位的功能嘛？冒昧的问一下！

**Reply by Waldenth** [reply]
> 大佬能在这个项目中实现虚拟定位的功能嘛？冒昧的问一下！

模拟定位避免检测的核心在于完全不干涉跑步App的所有交互，即手机运行跑步App的同时不断修改传感器记录的GPS坐标，让App误以为正在跑步。本项目只是在pc端模拟了app与服务器交互跑步成绩时的核心步骤，模拟定位无法发挥反检测作用，且没有必要。

**Reply by 18209618470** [reply]
但是模拟休眠就要不是跑不成功要不是进度条一直是白的没有进度。

**Reply by Waldenth** [reply]
> 但是模拟休眠就要不是跑不成功要不是进度条一直是白的没有进度。

抱歉，这应该是由于0.1.4-0.1.5版本的一个严重bug所造成的，该bug导致了模拟休眠模式运行出错，现在之前我已经修复了该bug并在readme中做了更新说明，你可以去下载修复后的0.1.5版本测试一下。
